misk-crypto/src/main/kotlin/misk/crypto/S3ExternalKeyManager.kt

```diff
@@ -32,21 +35,31 @@ import misk.logging.getLogger
 class S3ExternalKeyManager @Inject constructor(
   private val env: Env,
 
-  private val s3: AmazonS3,
+  defaultS3: AmazonS3,
 
   @ExternalDataKeys override val allKeyAliases: Map<KeyAlias, KeyType>,
 
   @Inject(optional = true)
   private val bucketNameSource: BucketNameSource = object : BucketNameSource {
     override fun getBucketName(env: Env) = env.name.toLowerCase()
-  }
-) : ExternalKeyManager {
+  },
 
-  private val logger by lazy { getLogger<S3ExternalKeyManager>() }
+  // The data keys bucket might live in a different region than the region this service executes
+  // in; if BucketNameSource provides a non-standard bucket region, we need to create a new S3
+  // client for that bucket, and to do that we need credentials again.
+  private val awsCredentials: AWSCredentialsProvider,
+) : ExternalKeyManager {
 
-  private val metadataKeyKmsArn = "kms-key-arn"
+  private val s3: AmazonS3 = bucketNameSource.getBucketRegion(env)?.let { region ->
+    logger.info("creating S3ExternalKeyManager S3 client for $region")
+    AmazonS3ClientBuilder
+      .standard()
+      .withRegion(region)
+      .withCredentials(awsCredentials)
+      .build()
+  } ?: defaultS3
 
-  private val metadataKeyKeyType = "key-type"
+  private val logger by lazy { getLogger<S3ExternalKeyManager>() }
```

Comment:
super nit: I think we usually prefer to initialize the logger object inside a `companion object {}` to make it static